modules/siddhi-core/src/main/java/org/wso2/siddhi/core/query/processor/stream/window/SessionWindowProcessor.java

```diff
@@ -60,37 +60,45 @@
 @Extension(
         name = "session",
         namespace = "",
-        description = "A session window that holds events into a session "
-                + "which has a grouping attribute (session key). After a session gap period, "
-                + "the session window would be expired. When a new event "
-                + "arrives with a value for session key, then it matches with a session window "
-                + "which has the same session key. A latency can be added to late arrival events "
-                + "to merge with the previous session windows. That latency time period "
-                + "should be less than the session gap time period. To have aggregate functions with session window, "
-                + "a 'group by' should be done with the session key. ",
+        description = "This is a session window that holds events that belong to a specific session. The events " +
+                "that belong to a specific session are identified by a grouping attribute (i.e., a session key). A " +
+                "session gap period is specified to determine the time period after which the session is considered " +
+                "to be expired. A new event that arrives with a specific value for the session key is matched with" +
+                " the session window with the same session key.\n " +
+                " When performing aggregations for a specific session, you can include events with the matching " +
+                "session key that arrive after the session is expired if required. This is done by specifying a " +
+                "latency time period that is less than the session gap period.\n" +
+                "To have aggregate functions with session windows, the events need to be grouped by the " +
+                "session key via a 'group by' clause.",
         parameters = {
                 @Parameter(name = "window.session",
-                        description = "The session gap time period(sec, min, ms).",
+                        description = "The time period for which the session considered is valid. This is specified" +
+                                " in seconds, minutes, or milliseconds.",
```

Comment:
can we also add (sec, min, ms) within brackets as well? So the user knows what are the values that can be used here?

Reply:
Did this update. Thank you.